modules/accounts/src/lib.rs

```diff
@@ -452,6 +421,26 @@ impl<T: Trait> Module<T> {
 			}
 		}
 	}
+
+	fn do_merge_account_check(source: &T::AccountId) -> DispatchResult {
+		// check must allow death,
+		// if native/non-native currencies has locks, means ref_count shouldn't be zero,
+		// can not close the account.
+		ensure!(<system::Module<T>>::allow_death(source), Error::<T>::NonZeroRefCount);
+
+		let native_currency_id = T::NativeCurrencyId::get();
+		let new_account_deposit = T::NewAccountDeposit::get();
```

Comment:
Add a comment to here `This cannot be changed without migration code to adjust reserved balances or update module_accounts::Module::do_merge_account_check`

https://github.com/AcalaNetwork/Acala/blob/5d218d9c397dfa272b0781be13893d8da16fd667/runtime/mandala/src/lib.rs#L1028